autorest/azure/azure.go

```diff
@@ -126,6 +131,58 @@ func ExtractRequestID(resp *http.Response) string {
 	return autorest.ExtractHeaderValue(HeaderRequestID, resp)
 }
 
+// GetAsyncOperation retrieves the long-running URL to poll from the passed response.
+func GetAsyncOperation(resp *http.Response) string {
+	return resp.Header.Get(http.CanonicalHeaderKey(HeaderAsyncOperation))
+}
+
+// ResponseIsLongRunning returns true if the passed response is for an Azure long-running operation.
+func ResponseIsLongRunning(resp *http.Response) bool {
+	return autorest.ResponseRequiresPolling(resp, http.StatusCreated) && len(GetAsyncOperation(resp)) > 0
```

Comment:
We must follow the specification. The last time I read it, it outlined the implemented behavior. If that's changed, I want to see the update before I make code changes.